packages/flutter/lib/src/widgets/focus_scope.dart

```diff
@@ -1,145 +1,422 @@
+/// This widget does not provide any visual indication that the focus has
+/// changed. Any desired visual changes should be made when [onFocusChanged] is
+/// called.
 ///
-/// ```dart
-/// FocusScope.of(context).setFirstFocus(node);
+/// To access the [FocusNode] of the [Focus] widget above this one in the widget
+/// hierarchy and establish a relationship that will rebuild the widget when
+/// the focus changes, use the [Focus.of] and [FocusScope.of] static methods.
+///
+/// To access the focused state of the nearest [Focus] widget, use
+/// [Focus.hasFocus] from a build method, which also establishes a relationship
+/// between the calling widget and the [Focus] widget that will rebuild the
+/// calling widget when the focus changes.
+///
+/// Managing a [FocusNode] means managing its lifecycle, listening for changes
+/// in focus, and re-parenting it when needed to keep the focus hierarchy in
+/// sync with the widget hierarchy. See [FocusNode] for more information about
+/// the details of what node management entails if not using a [Focus] widget.
+///
+/// To collect a sub-tree of nodes into a group, use a [FocusScope].
+///
+/// {@tool snippet --template=stateful_widget_material}
+/// This example shows how to manage focus using the [Focus] and [FocusScope]
+/// widgets. See [FocusNode] for a similar example that doesn't use [Focus] or
+/// [FocusScope].
+///
+/// ```dart imports
+/// import 'package:flutter/services.dart';
 /// ```
 ///
-/// If a [FocusScope] is removed from the widget tree, then the previously
-/// focused node will be focused, but only if the [node] is the same [node]
-/// object as in the previous frame. To assure this, you can use a GlobalKey to
-/// keep the [FocusScope] widget from being rebuilt from one frame to the next,
-/// or pass in the [node] from a parent that is not rebuilt. If there is no next
-/// sibling, then the parent scope node will be focused.
+/// ```dart
+/// Color _color = Colors.white;
+///
+/// bool _handleKeyPress(FocusNode node, RawKeyEvent event) {
+///   if (event is RawKeyDownEvent) {
+///     print('Focus node ${node.debugLabel} got key event: ${event.logicalKey}');
+///     if (event.logicalKey == LogicalKeyboardKey.keyR) {
+///       print('Changing color to red.');
+///       setState(() {
+///         _color = Colors.red;
+///       });
+///       return true;
+///     } else if (event.logicalKey == LogicalKeyboardKey.keyG) {
+///       print('Changing color to green.');
+///       setState(() {
+///         _color = Colors.green;
+///       });
+///       return true;
+///     } else if (event.logicalKey == LogicalKeyboardKey.keyB) {
+///       print('Changing color to blue.');
+///       setState(() {
+///         _color = Colors.blue;
+///       });
+///       return true;
+///     }
+///   }
+///   return false;
+/// }
+///
+/// @override
+/// Widget build(BuildContext context) {
+///   final TextTheme textTheme = Theme.of(context).textTheme;
+///   return FocusScope(
```

Comment:
Wondering while reading this example: Is the FocusScope actually required to make this work?

Reply:
No.  There's the top level focus scope provided by the manager, but I wanted to should how one would use a scope in context.